tests/integration/test_automation/testParser.py

```diff
@@ -0,0 +1,507 @@
+       file1 = open(i, 'r')
+       Lines = file1.readlines()
+       tr_ace(Lines,ace_master_3_write, 'AWSNOOP', 'write')       
+   for i in filenames_dict['ace_log/ccu_logger_snoop_0/snoop_read']:
+       file1 = open(i, 'r')
+       Lines = file1.readlines()
+       tr_snoop(Lines,  ace_master_0_snoop)     
+   for i in filenames_dict['ace_log/ccu_logger_snoop_1/snoop_read']:
+       file1 = open(i, 'r')
+       Lines = file1.readlines()
+       tr_snoop(Lines,  ace_master_1_snoop)      
+   for i in filenames_dict['ace_log/ccu_logger_snoop_2/snoop_read']:
+       file1 = open(i, 'r')
+       Lines = file1.readlines()
+       tr_snoop(Lines,  ace_master_2_snoop)    
+   for i in filenames_dict['ace_log/ccu_logger_snoop_3/snoop_read']:
+       file1 = open(i, 'r')
+       Lines = file1.readlines()
+       tr_snoop(Lines,  ace_master_3_snoop)       
+   for i in filenames_dict['axi_log/ccu_logger_slave/read']:
+       file1 = open(i, 'r')
+       Lines = file1.readlines()
+       tr_axi(Lines,   axi_slave_read , 'AR',  'read')        
+   for i in filenames_dict['axi_log/ccu_logger_slave/write']:
+       file1 = open(i, 'r')
+       Lines = file1.readlines()
+       tr_axi(Lines,   axi_slave_write , 'AW',  'write')
+
+def print_warnings():
+    for i in range(0,5):
+        log_file.write( "WARNING !! CRRESP bits checked as a 5 bit word. Not individually checked !!! \n")
+        log_file.write("WARNING !! TB only supports cacheline aligned accesses !!! \n")
+        log_file.write("WARNING !! TB ignores addressses in files not specified in Yaml via start/end indexso if Yaml does not include all addresses, it will still flag it as pass !!! \n")
+
+def are_there_any_extra_messages():
+
+  
+    is_there_extra_messages = False;
+    log_file.write("Checking for extra messages \n")
+    for theName, theObject in list_checker_dict.items():
+        log_file.write(theName + "\n")
+        if theObject.isListEmpty() != True:
+            log_file.write(theName + " has extra messages \n" )
+            is_there_extra_messages = True
+
+
+    return is_there_extra_messages        
+
+    
+
+if __name__ == "__main__":
+   buffer_access_dict = {}
+   filenames_dict = {}
+   #arsnoop dictionary defs from from IHI0022H_c_amba_axi_protocol_spec.pdf table D3.7
+   arsnoop_dict = {'ReadNoSnoop':0,'ReadOnce':0,'ReadShared':1,'ReadClean':1,'ReadNotSharedDirty':3,'ReadUnique':7,'CleanUnique':11,'CleanShared':8,'CleanInvalid':9,'MakeInvalid':13}
+   #awsnoop dictionary defs from from IHI0022H_c_amba_axi_protocol_spec.pdf table D3.8
+   awsnoop_dict = {'WriteNoSnoop':0,'WriteUnique':0,'WriteLineUnique':1,'WriteClean':2,'WriteBack':3,'Evict':4,'WriteEvict':5}
+   #acsnoop dictionary defs from from IHI0022H_c_amba_axi_protocol_spec.pdf table D3.19
+   acsnoop_dict = {'ReadOnce':0,'ReadShared':1,'ReadClean':2,'ReadNotSharedDirty':3,'ReadUnique':7,'CleanShared':8,'CleanInvalid':9}   
+   ace_master_0_read = []   
```

Comment:
could be 2D array

Reply:
not sure I agree here.  The values are not linear, and in the case of the awsnoop, two keys have the value 0.

Reply:
Maybe you are saying that each dictionary can be inserted into another dictionary distinguished by a key

Reply:
I am referring to the lists, e.g. `ace_master_x_read`

Reply:
It goes together with this comment https://github.com/planvtech/culsans/pull/10/files#r1195031312 about using a loop

Reply:
okay

Reply:
see below